### PR TITLE
Update configuration file logic for the analytic global option

### PR DESCRIPTION
**Pull request by antalszava**
**Context:**
A configuration file can store details that will be used for every loaded device in PennyLane such as  the value for the `analytic` device attribute. However, certain devices do not support the `analytic` attribute upon instantiation. Therefore an error is raised when they are instantiated with `qml.device()`.

Examples of these device are the `default.tensor.tf` and `forest.qvm` devices (`'QVM device cannot be run in analytic=True mode.`).

**Description of the Change:**
Added logic that checks if the device accepts an `analytic` argument or not. 

**Benefits:**
The `analytic` attribute can further be supported by the configuration file. An alternative solution would entail removing it as is.

**Possible Drawbacks:**
The check assumes that the parameter named `analytic` appears in the signature of the `__init__` method of the specific Device.

**Related GitHub Issues:**
N/A

### CODE REVIEW

File: pennylane/__init__.py [hunk 1]
```diff
@@ -15,35 +15,33 @@
...
+from ._version import __version__
+from .circuit_graph import CircuitGraph
+from .collections import QNodeCollection, apply, dot, map, sum
+from .configuration import Configuration
+from .io import *
+from .measure import expval, probs, sample, var
 from .ops import *
 from .optimize import *
-from .qnodes import qnode, QNode, QuantumFunctionError
+from .qnodes import QNode, QuantumFunctionError, qnode
```
Comment (josh146) [inline pennylane/__init__.py]: I'm assuming the above was because of isort? 😆

Comment (antalszava) [reply pennylane/__init__.py]: Yes! :smile:

**Comment by antalszava** [comment]
The reason for the changes is that `analytic=True` is set in the example [configuration file on the website](https://pennylane.readthedocs.io/en/stable/introduction/configuration.html#format).

My initial thought was to just simply remove it, but then it came to me that we could include a logic that bypasses it.

Overall, I feel it would behttps://pennylane.readthedocs.io/en/stable/introduction/configuration.html#formatW

**Comment by antalszava** [comment]
The reason for the changes is that `analytic=True` is set in the example [configuration file on the website](https://pennylane.readthedocs.io/en/stable/introduction/configuration.html#format). So overall using this example config file and using a hardware device would raise an error.

My initial thought was to just simply remove it, but then it came to me that we could include a logic that bypasses it.

Let me know which one would be preferable!

(The basic scenario that I would like to avoid is that a user creates the configuration file, uses simulator devices and then _forgets_ that the `analytic` attribute was globally set. Then at a later stage when they get the related error using a hardware device, they might only be looking in their code ultimately skipping the configuration file.)

**Comment by josh146** [comment]
> The reason for the changes is that analytic=True is set in the example configuration file on the website. So overall using this example config file and using a hardware device would raise an error.
>
> My initial thought was to just simply remove it, but then it came to me that we could include a logic that bypasses it.

Ah, I see! In this case, perhaps we should just remove it from the documentation example? I find it a bit confusing to have logic that bypasses a configuration option. Perhaps the best solution is to update the error message itself.

**Comment by antalszava** [comment]
Updated this such that only the configuration file is modified! Was not too sure about your comment on updating the error message, could you elaborate? :)

**Pull request merged**